hack/import-restrictions.json

```diff
@@ -390,5 +390,168 @@
       "github.com/openshift/origin/pkg/user/registry/useridentitymapping",
       "github.com/openshift/origin/pkg/util/httprequest"
     ]
+  },
+
+  {
+    "checkedPackageRoots": [
+      "github.com/openshift/origin/pkg/oc"
+    ],
+    "allowedImportPackageRoots": [
+      "vendor/github.com/aws/aws-sdk-go",
+      "vendor/github.com/blang/semver",
+      "vendor/github.com/containers/image",
+      "vendor/github.com/daviddengcn/go-colortext",
+      "vendor/github.com/docker/distribution",
+      "vendor/github.com/docker/docker",
+      "vendor/github.com/docker/engine-api",
+      "vendor/github.com/docker/go-connections",
+      "vendor/github.com/docker/go-units",
+      "vendor/github.com/evanphx/json-patch",
+      "vendor/github.com/fsnotify/fsnotify",
+      "vendor/github.com/fsouza/go-dockerclient",
+      "vendor/github.com/golang/glog",
+      "vendor/github.com/gonum/graph",
+      "vendor/github.com/joho/godotenv",
+      "vendor/github.com/MakeNowJust/heredoc",
+      "vendor/github.com/opencontainers/go-digest",
+      "vendor/github.com/openshift/api",
+      "vendor/github.com/openshift/client-go",
+      "vendor/github.com/openshift/imagebuilder",
+      "vendor/github.com/openshift/source-to-image",
+      "vendor/github.com/pborman/uuid",
+      "vendor/github.com/prometheus/client_golang",
+      "vendor/github.com/RangelReale/osincli",
+      "vendor/github.com/spf13/cobra",
+      "vendor/github.com/spf13/pflag",
+      "vendor/golang.org",
+      "vendor/gopkg.in/ldap.v2",
+      "vendor/gopkg.in/yaml.v2",
+      "vendor/k8s.io/api",
+      "vendor/k8s.io/apimachinery",
+      "vendor/k8s.io/apiserver",
+      "vendor/k8s.io/client-go",
+      "vendor/k8s.io/kube-aggregator",
+      "vendor/k8s.io/kubernetes",
+      "vendor/k8s.io/utils",
+
+      "github.com/openshift/origin/pkg/apps/generated",
+      "github.com/openshift/origin/pkg/authorization/generated",
+      "github.com/openshift/origin/pkg/build/generated",
+      "github.com/openshift/origin/pkg/image/generated",
+      "github.com/openshift/origin/pkg/network/generated",
+      "github.com/openshift/origin/pkg/oauth/generated",
+      "github.com/openshift/origin/pkg/project/generated",
+      "github.com/openshift/origin/pkg/quota/generated",
+      "github.com/openshift/origin/pkg/route/generated",
+      "github.com/openshift/origin/pkg/user/generated",
+      "github.com/openshift/origin/pkg/security/generated",
+      "github.com/openshift/origin/pkg/template/generated"
+    ],
+    "allowedImportPackages": [
```

Comment:
@soltysh I'm not saying to abandon all hope, but it's looking pretty tough.

Reply:
Sigh...